cognite/neat/data_model/models/dms/_constraints.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC
+from typing import Annotated, Literal
+
+from pydantic import Field
+
+from ._base import BaseModelObject
+from ._references import ContainerReference
+
+
+class ConstraintDefinition(BaseModelObject, ABC):
+    constraint_type: str
+
+
+class UniquenessConstraintDefinition(ConstraintDefinition):
+    constraint_type: Literal["uniqueness"] = "uniqueness"
+    properties: list[str] = Field(
+        description="List of properties included in the constraint.", min_length=1, max_length=10
+    )
+    by_space: bool | None = Field(default=None, description="Whether to make the constraint space-specific.")
+
+
+class RequiresConstraintDefinition(ConstraintDefinition):
+    constraint_type: Literal["requires"] = "requires"
+    require: ContainerReference = Field(description="Reference to an existing container.")
+
+
+Constraint = Annotated[
+    UniquenessConstraintDefinition | RequiresConstraintDefinition,
+    Field(discriminator="constraint_type"),
+]
```

Comment:
This pattern is used for indexes and constraints as well and I intend to use it through the new pydantic classes. 

The downside of it is that for every new constraint, you have to remember to update the Union here. I tried multiple other approaches that used reflections, however, none became as clean as this. In addition, the solution has to be compatible with `pydantic` 2.10.6 for Neat to be compatible with `pyodide`. Thus, I cannot use some of the wrappers that was introduced later.